src/Service/SigningService.php

```diff
@@ -0,0 +1,362 @@
+      $this->debug('Moving %filename to %signing_pdf.', ['%filename' => $this->getUploadFilesDir() . $filename, '%signing_pdf' => $signingPdf]);
+      $this->fileSystem->move($this->getUploadFilesDir() . '/' . $filename, $signingPdf);
+    }
+
+    $cid = $this->getCorrelationId();
+    $serviceUrl = $this->config->get('signing_service_url');
+    if (!$serviceUrl) {
+      throw new SigningException('Signing service URL not set. Please set it in the settings form.');
+    }
+
+    $url = $serviceUrl . '/sign/pdf/' . basename($signingPdf) . '?' . http_build_query(['correlationId' => $cid]);
+
+    $this->debug('Signing file %url', ['%url' => $url]);
+
+    return new TrustedRedirectResponse($url);
+  }
+
+  /**
+   * Result callback.
+   *
+   * User has returned from a signing service, clearing cookies, redirecting
+   * to the forward url.
+   *
+   * @params string $file
+   *   File name.
+   * @oaram string $action
+   *   Expected action is 'result' or 'cancel'.
+   *
+   * @return TrustedRedirectResponse
+   *   Redirect response.
+   */
+  public function result(string $file, string $action) : TrustedRedirectResponse {
+    $request = $this->requestStack->getCurrentRequest();
+
+    if (empty($cookie = $request->cookies->get(SigningService::FORWARD_URL_COOKIE) ?? NULL)) {
+      throw new AccessDeniedHttpException('Required cookie not found.');
+    }
+
+    $url = base64_decode($cookie);
+    if (!$this->isValidUrl($url)) {
+      throw new AccessDeniedHttpException("Provided cookie forward url value not accepted: $url");
+    }
+
+    $sep = str_contains($url, '?') ? '&' : '?';
+    $response = new TrustedRedirectResponse($url . $sep . http_build_query(['file' => $file, 'action' => $action]));
+
+    // Deleting cookies.
+    $response->headers->clearCookie(SigningService::FORWARD_URL_COOKIE);
+
+    return $response;
+  }
+
+  /**
+   * Download the signed file.
+   */
+
+  /**
+   * Sends the file as a binary response.
+   *
+   * @param string $file
+   *   File to attempt download for.
+   * @param bool $leave
+   *   If a file should be left on the server.
+   *   TRUE - file remains untouched.
+   *   FALSE - file will be deleted after sending.
+   * @return BinaryFileResponse
+   *   Found a file.
+   */
+  public function download(string $file, bool $leave) : BinaryFileResponse {
+    if (!preg_match('/^[a-z0-9]{32}\.pdf$/', $file)) {
+      throw new BadRequestHttpException("Invalid file name: $file. Must be contain letters or numbers and be 32 chars long");
+    }
+
+    $signedPdf = $this->getSignedFilesDir() . '/' . substr($file, 0, 32) . '-signed.pdf';
+
+    $filesize = (int) @filesize($signedPdf);
+    $this->debug('Sending %signed_pdf (%filesize bytes)', ['%signed_pdf' => $signedPdf, '%filesize' => $filesize]);
+    if ($filesize) {
+      $response = new BinaryFileResponse($signedPdf);
+      $response->headers->set('Content-Type', 'application/pdf');
+      $response->headers->set('Content-Length', $filesize);
```

Comment:
Doesn't `BinaryFileResponse` set this header automatically?

Reply:
i didn't find it is doing this. Am i missing anything?